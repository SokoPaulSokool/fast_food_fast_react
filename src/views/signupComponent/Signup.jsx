import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import { connect } from 'react-redux';
import { MDBCard, MDBBtn, MDBInput, MDBCardBody } from 'mdbreact';
import { signUpAction } from '../../actions/authAction';
import './Signup.scss';

export class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = { errors: { message: '' } };
  }

  componentWillReceiveProps(nextProps) {
    const { errors, user } = nextProps;
    const { history } = this.props;
    if (errors) {
      this.setState({ errors: errors });
    }
    if (user.user_name) {
      toast.success('Sign up was successful. You can now login', {
        position: toast.POSITION.TOP_CENTER
      });
      setTimeout(() => {
        history.push('./');
      }, 3000);
    }
  }
  submit = e => {
    e.preventDefault();
    const { user_name, email, password } = this.state;
    const { signUpAction } = this.props;
    const data = {
      user_name: user_name,
      email: email,
      password: password,
      account_type: 'customer'
    };
    signUpAction(data);
  };
  onChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };
  render() {
    const { errors } = this.state;
    return (
      <div className="container-fluid  mt-5 col-md-6">
        <MDBCard className="dark-grey-text">
          <MDBCardBody className="z-depth-2">
            <h3 className="dark-grey-text text-center">
              <strong>Sign Up</strong>
            </h3>
            <hr />
            <form>
              <MDBInput
                name="user_name"
                onChange={this.onChange}
                label="Your name"
                icon="user"
              />
              <MDBInput
                onChange={this.onChange}
                name="email"
                label="Your email"
                icon="envelope"
              />
              <MDBInput
                onChange={this.onChange}
                name="password"
                type="password"
                label="Your password"
                icon="lock"
              />
              <p className="red-text">{errors.message}</p>

              <div className="text-center mt-3 black-text">
                <MDBBtn
                  type="submit"
                  onClick={this.submit}
                  className="bg-secondary"
                >
                  Submit
                </MDBBtn>
                <hr />
              </div>
            </form>
          </MDBCardBody>
        </MDBCard>
      </div>
    );
  }
}
Signup.propTypes = {
  signUpAction: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
  errors: PropTypes.shape({}),
  user: PropTypes.shape({})
};

Signup.defaultProps = {
  errors: { message: '' },
  user: {}
};
const mapStateToProps = state => {
  return { user: state.auth.user, errors: state.auth.errors };
};

export default connect(
  mapStateToProps,
  { signUpAction }
)(Signup);
