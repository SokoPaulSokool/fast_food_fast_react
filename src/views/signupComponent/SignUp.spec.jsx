import React from 'react';
import { shallow } from 'enzyme';

import { Signup } from './Signup';

function setup() {
  const props = {
    signUpAction: jest.fn,
    history: {}
  };
  const wrapper = shallow(<Signup {...props} />);
  return { wrapper, props };
}

describe('First test', () => {
  it('matches snapshort', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call handleSubmit ', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    expect(
      instance.submit({
        target: { value: 'tag' },
        preventDefault: jest.fn
      })
    );
  });
  it('should call onChange ', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    expect(
      instance.onChange({
        target: { value: 'tag' },
        preventDefault: jest.fn
      })
    );
  });
  it('Test component did mount', () => {
    const { wrapper } = setup();
    wrapper.setProps({ auth: { user: { user_name: 'soko' } } });
    expect(wrapper.state()).toEqual({ errors: { message: '' } });
  });
  it('Test component did mount', () => {
    const { wrapper } = setup();
    wrapper.setProps({ auth: { errors: { message: 'empty' } } });
    expect(wrapper.state()).toEqual({ errors: { message: '' } });
  });
  it('Should have MDBBtn', () => {
    const { wrapper } = setup();
    expect(wrapper.find('div').exists()).toBe(true);
  });
});
