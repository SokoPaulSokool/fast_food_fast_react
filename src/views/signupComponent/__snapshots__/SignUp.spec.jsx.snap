// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First test matches snapshort 1`] = `ShallowWrapper {}`;
